File: Практика_самостоятельная работа/C++/4.3.cpp
#include <iostream>
#include <cmath>

using namespace std;

int main() {
    double x, y;

    cout << "Введите x: ";
    cin >> x;

    if (x > 0) {
        y = sin(x * x);
    } else {
        y = 1 - 2 * pow(sin(x), 2);
    }

    cout << "y = " << y << endl;

    return 0;
}

File: Практика_самостоятельная работа/Python/4.3.py
import math

# Ввод значения x
x = float(input("Введите x: "))

# Расчёт значения y
if x > 0:
    y = math.sin(x**2)
else:
    y = 1 - 2 * math.sin(x)**2

# Вывод результата
print("y =", y)
